### ISSUE TITLE
`Next` cop throwing false negative with complex conditionals

### ISSUE THREAD

**Issue by patbenatar** [rubocop/rubocop#1225]
I'm getting the "Use next to skip iteration." error for the following code:

``` ruby
variant_set.each_pair do |variant, desired_quantity|
  if source
    actual_quantity = unstock_source(variant, desired_quantity)
  else
    actual_quantity = desired_quantity
  end

  if destination
    destination.add_stock(variant, actual_quantity, inventory_transfer)
  end

  if actual_quantity < desired_quantity
    remainder_set[variant] = desired_quantity - actual_quantity
  end
end
```

AFAIK this is a false negative. Am I missing something?
